### PR TITLE
Lab1: wav processor

### PR DESCRIPTION
**Pull request by insouciiance**

### CODE REVIEW

File: Lab_1/IP-01_Kovalenko_Mykyta/.project [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ohorielov) [inline Lab_1/IP-01_Kovalenko_Mykyta/.project]: remove this file

File: Lab_1/IP-01_Kovalenko_Mykyta/.gitignore [hunk 2]
```diff
@@ -0,0 +1,60 @@
+.metadata
```
Comment (ohorielov) [inline Lab_1/IP-01_Kovalenko_Mykyta/.gitignore]: remove this file

File: Lab_1/IP-01_Kovalenko_Mykyta/src/wav_header.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	uint16_t num_channels;
+	uint32_t sample_rate;
+	uint32_t byte_rate;
+	uint16_t block_align;
+	uint16_t bits_per_sample;
+
+	uint32_t subchunk2_id;
+	uint32_t subchunk2_size;
+
+	int16_t *data;
```
Comment (ohorielov) [inline Lab_1/IP-01_Kovalenko_Mykyta/src/wav_header.h]: read about variable-length arrays and replace this by  uint16_t data[], also is unsigned, so int16_t is not right representation

File: Lab_1/IP-01_Kovalenko_Mykyta/src/wav_processor.c [hunk 4]
```diff
@@ -0,0 +1,82 @@
+#include "stdlib.h"
+#include "stdio.h"
+#include "stdint.h"
+#include "wav_header.h"
+#include "wav_processor.h"
+
+void load_wav(const char *filename, wav_header *wav) {
+	FILE *file = fopen(filename, "rb");
+
+	fread(&wav->chunk_id, 4, 1, file);
```
Comment (ohorielov) [inline Lab_1/IP-01_Kovalenko_Mykyta/src/wav_processor.c]: magic numbers, use #define to declare all constants

File: Lab_1/IP-01_Kovalenko_Mykyta/src/wav_processor.c [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+	wav->data = malloc(wav->subchunk2_size);
+	fread(wav->data, wav->subchunk2_size, 1, file);
+
+	fclose(file);
+}
+
+void write_wav(const char *filename, wav_header *wav) {
+	FILE* file = fopen(filename, "wb");
+
+	fwrite(&wav->chunk_id, 4, 1, file);
```
Comment (ohorielov) [inline Lab_1/IP-01_Kovalenko_Mykyta/src/wav_processor.c]: you can create fulfilled structure and write it by single function call

Comment (ohorielov) [reply Lab_1/IP-01_Kovalenko_Mykyta/src/wav_processor.c]: also you can use header from read file, they will be the same

File: Lab_1/IP-01_Kovalenko_Mykyta/src/wav_processor.c [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+	fclose(file);
+}
+
+void dump_wav(wav_header *wav) {
+	printf("Chunk ID: %c%c%c%c\n", wav->chunk_id & 0xFF, (wav->chunk_id >> 8) & 0xFF, (wav->chunk_id >> 16) & 0xFF, (wav->chunk_id >> 24) & 0xFF);
+	printf("Chunk size: %u\n", wav->chunk_size);
+	printf("Format: %c%c%c%c\n", wav->format & 0xFF, (wav->format >> 8) & 0xFF, (wav->format >> 16) & 0xFF, (wav->format >> 24) & 0xFF);
+
+	printf("Subchunk 1 ID: %c%c%c%c\n", wav->subchunk1_id & 0xFF, (wav->subchunk1_id >> 8) & 0xFF, (wav->subchunk1_id >> 16) & 0xFF, (wav->subchunk1_id >> 24) & 0xFF);
```
Comment (ohorielov) [inline Lab_1/IP-01_Kovalenko_Mykyta/src/wav_processor.c]: calculations in printf is not appreciated

File: Lab_1/IP-01_Kovalenko_Mykyta/src/wav_processor.c [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+	printf("Sample rate: %u\n", wav->sample_rate);
+	printf("Byte rate: %u\n", wav->byte_rate);
+	printf("Block align: %hu\n", wav->block_align);
+	printf("Bits per sample: %hu\n", wav->bits_per_sample);
+
+	printf("Subchunk 2 ID: %c%c%c%c\n", wav->subchunk2_id & 0xFF, (wav->subchunk2_id >> 8) & 0xFF, (wav->subchunk2_id >> 16) & 0xFF, (wav->subchunk2_id >> 24) & 0xFF);
+	printf("Subchunk 2 size: %u\n", wav->subchunk2_size);
+}
+
+void strengthen_wav(wav_header *wav, float scale) {
```
Comment (ohorielov) [inline Lab_1/IP-01_Kovalenko_Mykyta/src/wav_processor.c]: what means this fcn name?

File: Lab_1/IP-01_Kovalenko_Mykyta/src/wav_processor.c [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+	int samples_count = wav->subchunk2_size / sizeof(int16_t);
+
+	for(int i = 0; i < samples_count; i++) {
```
Comment (ohorielov) [inline Lab_1/IP-01_Kovalenko_Mykyta/src/wav_processor.c]: use:
uint32_t i;
for(i = 0; i < samples_cnt; i++)

File: Lab_1/IP-01_Kovalenko_Mykyta/src/main.c [hunk 9]
```diff
@@ -0,0 +1,14 @@
+#include "stdlib.h"
+#include "stdio.h"
+#include "wav_processor.h"
+
+int main(void) {
+	wav_header *header = malloc(sizeof *header);
+	load_wav("../Master Of Puppets.wav", header);
+	dump_wav(header);
+	strengthen_wav(header, 0.75);
```
Comment (ohorielov) [inline Lab_1/IP-01_Kovalenko_Mykyta/src/main.c]: calculate 3 dB int code, please

**Comment by insouciiance** [comment]
Please review my fixes @ohorielov

**Pull request merged**